terraform/README.md

```diff
@@ -60,8 +60,38 @@ For these commands ensure you are in the terraform initialised folder when runni
 ## How it works
 #### main.tf
 - This contains the terraform config required for provisioning the cloud architecture.
+- Existing resources are specified using the `data` keyword they include:
+    - VPC - Virtual private cloud.
+    - Public Subnet - a subsection of the VPC that is accessible externally.
+    - ECS cluster - this is the cluster of machines our dashboard will run on.
+- Local variables are defined by the `locals` keyword this defines:
+    - API keys - for running the lambda pipelines.
+    - Database variables - to securely access the RDS instance.
+    - Image URIs - for the ECR repositories to be used.
+- New resources are defined by the `resource` keyword and these services can be grouped into the following:
+    - RDS:
+        - The RDS instance - the database itself.
+        - Database subnet group - the specific subnet the RDS will run on.
+        - Database security group - the rules of accessing the database.
+    - Lambda:
+        - The lambda function - created by recursion through the use of a `local` block.
+        - AWS cloudwatch rules - allows for logging.
+        - IAM role - an administration role that allows the lambdas to execute and to pull images from ECR. These are assigned by recursion and a `if` statement so that only the minimum required permisions are granted.
+    - State machine: 
+        - Step function - creates two step functions that target the lambdas to run either hourly or quadhorally.
+        - IAM role - an administration role that allows the state machine to execute.
+    - ECS:
+        - IAM role - an administration role that allows the ECS task to execute.
+        - AWS cloudwatch rules - allows for logging by ecs.
+        - ECS task definition - this defines which ECR image will be loaded and on what machine.
+        - ECS service - a system that restarts the ECS task upon failure.
+    - Eventbridge:
+        - Cloudwatch roles - allows for logging by eventbridge
+        - Eventbridge bus - A quadhoral, hourly and weekly eventbridge triggers are provisioned
+
 #### variables.tf
```

Comment:
tf.vars of course in the environment variables / .env equivalent for Terraform, the main.tf file needs these variables, so it's good to mention for someone unfamiliar with Terraform.